rasa/core/policies/rule_policy.py

```diff
@@ -519,6 +520,25 @@ def _handling_loop_sources(domain: Domain) -> Set[Text]:
             )
         return loop_sources
 
+    def _remove_action_listen_contradiction(self, prediction_source: Text) -> None:
+        prev_action = json.loads(prediction_source)[-1][PREVIOUS_ACTION][ACTION_NAME]
```

Comment:
I don't think you need to do that `prediction_source` should be identical to one of the keys in rules lookup